src/Vocup/MainForm.cs

```diff
@@ -125,9 +125,63 @@ public void UnloadBook(bool fullUnload)
             }
         }
 
+        /// <summary>
+        /// Save the location, state and size of this form.
+        /// </summary>
+        private void StoreSettings()
+        { 
+            Settings.Default.MainFormLocation    = RestoreBounds.Location;
+            Settings.Default.MainFormWindowState = WindowState;
+            Settings.Default.MainFormSize        = RestoreBounds.Size;
+
+            Settings.Default.Save();
+        }
+
+        /// <summary>
+        /// Restore the saved location, state and size of this form.
+        /// </summary>
+        private void RestoreSettings()
+        {
+            Location = Settings.Default.MainFormLocation;
+            
+            // Do not restore the window state when the form was minimzed
+            if (Settings.Default.MainFormWindowState != FormWindowState.Minimized)
+            {
+                WindowState = Settings.Default.MainFormWindowState;
+            }
+
+            if (Settings.Default.MainFormSize != default)
+            {
+                Size = Settings.Default.MainFormSize;
+            }
+
+            Location = Settings.Default.MainFormLocation;
+
+            // check if form is visible on any screen 
+            bool isVisible = false;
+            foreach (Screen screen in Screen.AllScreens)
+            {
+                if (screen.Bounds.IntersectsWith(this.Bounds))
+                {
+                    isVisible = true;
+                    break;
+                }
+            }
+
+            if (!isVisible)
```

Comment:
This check is really great, especially for people who change their screen setup sometimes. However, I would perform this check before restoring any settings and restore them only if it would still be inside the bounds. That means we would only restore settings if they are initialized and also inside the screens' bounds. Whether you restore the window's bounds or its state first doesn't make a difference but I would prefer restoring the bounds first.